canonicaljson.py

```diff
@@ -51,19 +40,35 @@ def _default(obj):
 # (in any case, simplejson's ensure_ascii doesn't get U+2028 and U+2029 right,
 # as per https://github.com/simplejson/simplejson/issues/206).
 #
-_canonical_encoder = json.JSONEncoder(
-    ensure_ascii=True,
-    separators=(',', ':'),
-    sort_keys=True,
-    default=_default,
-)
-
-_pretty_encoder = json.JSONEncoder(
-    ensure_ascii=True,
-    indent=4,
-    sort_keys=True,
-    default=_default,
-)
+
+# Declare these in the module scope, but they get configured in
+# set_json_library.
+_canonical_encoder = None
+_pretty_encoder = None
+
+
+def set_json_library(json_lib):
+    """
+    Set the encoders for canonicaljson to json_lib.
+
+    Params:
+        json_lib: The module to use for JSON encoding, Must have a
+            `JSONEncoder` property.
+    """
+    globals()["_canonical_encoder"] = json_lib.JSONEncoder(
```

Comment:
Is there a reason why the `global` statement isn't being used?

Reply:
Thanks for your feedback, but I'm still experimenting in this branch. I've reverted this PR back to a draft, which is what I meant to use in the first place.